File: verify_py1.0/tranfail/CalculateTheta.py
# coding: utf-8
from numpy import loadtxt, savetxt
from numpy import mean, shape, transpose
from numpy import matrix
from oneVsAll import oneVsAll
from predictOneVsAll import predictOneVsAll

def ccTheta():
    """
        测试成功
    """

    input_layer_size  = 324
    num_labels = 36

    X = matrix(loadtxt('./SampleDATA/X.dat'))
    y = matrix(loadtxt('./SampleDATA/y.dat')).transpose()
    the_lambda = 0.1

    all_theta = oneVsAll(X, y, num_labels, the_lambda)
    savetxt('theta.dat',all_theta)

    Accuracy, pred = predictOneVsAll(all_theta, X)

    print '\nTraining Set Accuracy: \n',   pred == y

    return all_theta

File: Verify-Manual-octave/train/core/CalculateTheta.m
clear ; close all; clc

input_layer_size  = 324;  % 27*12 Input Images of Digits
num_labels = 36;          % 10 labels, from 1 to z  

fprintf('Loading and Visualizing Data ...\n')

load ../samples/Xdata/X.dat;
load ../samples/Ydata/y.dat


lambda = 0.1;
[all_theta] = oneVsAll(X, y, num_labels, lambda);

save('theta.dat','all_theta')

pred = predictOneVsAll(all_theta, X);

fprintf('\nTraining Set Accuracy: %f\n', mean(double(pred == y)) * 100);
